service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/content/ReturnedCaseContentProvider.java

```diff
@@ -2,41 +2,62 @@
     private final ObjectMapper mapper;
-    private final LocalAuthorityNameLookupConfiguration localAuthorityNameLookupConfiguration;
+    private final LocalAuthorityNameLookupConfiguration localAuthorityNameLookup;
+    private final DocumentDownloadService documentDownloadService;
 
-    public ReturnedCaseTemplate buildNotificationParameters(CaseDetails caseDetails, String localAuthority) {
+    public ReturnedCaseTemplate parametersWithCaseUrl(CaseDetails caseDetails) {
         CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
 
-        ReturnApplication returnApplication = caseData.getReturnApplication();
-        ReturnedCaseTemplate returnedCaseTemplate = new ReturnedCaseTemplate();
+        return templateData(caseData)
+            .caseUrl(getCaseUrl(caseDetails.getId()))
+            .build();
+    }
 
-        returnedCaseTemplate.setLocalAuthority(
-            localAuthorityNameLookupConfiguration.getLocalAuthorityName(localAuthority));
+    public ReturnedCaseTemplate parametersWithApplicationLink(CaseDetails caseDetails) {
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
 
-        returnedCaseTemplate.setRespondentFullName(getFirstRespondentFullName(caseData.getRespondents1()));
-        returnedCaseTemplate.setRespondentLastName(getFirstRespondentLastName(caseData.getRespondents1()));
-        returnedCaseTemplate.setFamilyManCaseNumber(defaultIfNull(caseData.getFamilyManCaseNumber(), ""));
-        returnedCaseTemplate.setReturnedReasons(returnApplication.getFormattedReturnReasons());
-        returnedCaseTemplate.setReturnedNote(returnApplication.getNote());
-        returnedCaseTemplate.setCaseUrl(getCaseUrl(caseDetails.getId()));
+        return templateData(caseData)
+            .applicationDocumentUrl(linkToAttachedDocument(caseData.getSubmittedForm()))
+            .build();
+    }
+
+    private ReturnedCaseTemplate.ReturnedCaseTemplateBuilder templateData(CaseData caseData) {
+        ReturnApplication returnApplication = caseData.getReturnApplication();
+
+        return ReturnedCaseTemplate.builder()
+            .localAuthority(localAuthorityNameLookup.getLocalAuthorityName(caseData.getCaseLocalAuthority()))
+            .respondentFullName(getFirstRespondentFullName(caseData.getRespondents1()))
+            .respondentLastName(getFirstRespondentLastName(caseData.getRespondents1()))
+            .familyManCaseNumber(defaultIfNull(caseData.getFamilyManCaseNumber(), ""))
+            .returnedReasons(returnApplication.getFormattedReturnReasons())
+            .returnedNote(returnApplication.getNote());
+    }
 
-        return returnedCaseTemplate;
+    private Map<String, Object> linkToAttachedDocument(final DocumentReference documentReference) {
+        byte[] content = documentDownloadService.downloadDocument(documentReference.getBinaryUrl());
+        return generateAttachedDocumentLink(content).map(JSONObject::toMap).orElse(null);
```

Comment:
Are we silently ignoring when there are issues with documents? That makes me a bit worried.